src/utils.jl

```diff
@@ -614,4 +614,20 @@ macro refargs(ex)
         pushfirst!(ex.args[2].args, refs)
         return esc(ex)
     end
-end
\ No newline at end of file
+end
+
+# https://github.com/JuliaLang/julia/issues/40626
+# as suggested in the above issue, spawned tasks may
+# end up getting stuck in thread local storage
+# running clear_thread_states clears out any thread local storage tasks
+struct _Returns{V} <: Function
```

Comment:
why do you define `_Returns` instead of using `Base.Returns`?

Reply:
I ran into compat issues with 1.6; maybe I just needed to do `Base.Returns` instead and it was exported starting in 1.7?

Reply:
Ah yes, docs states:

> Returns requires at least Julia 1.7.